cub/cub/device/dispatch/dispatch_reduce.cuh

```diff
@@ -758,34 +761,51 @@ struct DispatchSegmentedReduce
         break;
       }
 
+      const auto num_segments_per_invocation =
+        static_cast<::cuda::std::int64_t>(::cuda::std::numeric_limits<std::int32_t>::max());
```

Comment:
```suggestion
        static_cast<::cuda::std::int64_t>(::cuda::std::numeric_limits<::cuda::std::int32_t>::max());
```

Reply:
Thanks! I'll update the branch once the rest of the review is in.